### PR TITLE
Fix styling throughout app and migrate to Less.js

### PR DESCRIPTION
**Pull request by andrewzulaybar**
- Converted all .css files to .less throughout app
- Removed native browser styling manipulations on Home page and just use flexbox

Note: still have to test out styling with promotion and restaurant cards

### CODE REVIEW

File: src/services/PromotionService.ts
Comment (ericrkuo) [inline src/services/PromotionService.ts:135]: The sort by recency seems to be sorting the wrong direction (showing the oldest ones at the top)

File: src/components/navigation/NavigationBar.tsx
Comment (ericrkuo) [inline src/components/navigation/NavigationBar.tsx:63]: Not related to this code I highlighted, but the file itself

I kind of like it if `my account` is the very right, I feel like in websites its not common for your account details to be one of the main ones (b/c I read left to right for the tabs and left-er ones seem more important).
image

File: src/components/navigation/NavigationBar.tsx
Comment (ericrkuo) [inline src/components/navigation/NavigationBar.tsx:52]: If you make the window **width** smaller, it collapses the menu items. That's correct, but if I then make the width larger, they do no uncollapse (expand). Is this happening for you too?

File: src/components/account/RegisterCard.tsx
Comment (ericrkuo) [inline src/components/account/RegisterCard.tsx]: Currently broken, needs to be:

```suggestion
          <Form.Item name="firstName" rules={InputRules.firstName} hasFeedback={true}>
            <Input placeholder="First name" />
          </Form.Item>
        </Col>
        <Col span={12}>
          <Form.Item name="lastName" rules={InputRules.lastName} hasFeedback={true}>
            <Input placeholder="Last name" />
          </Form.Item>
```

File: src/screens/MyPromotions.tsx
Comment (ericrkuo) [inline src/screens/MyPromotions.tsx:17]: Hmm, this is because the BE doesn't tell you vote state nor isSaved by for the uploadedPromotions endpoint... i don't think I can get to it on time ):


image

File: src/components/PromotionList.tsx
Comment (ericrkuo) [inline src/components/PromotionList.tsx:27]: I just realized we don't show the discount for a promotion, did something happen?
image

File: src/services/PromotionService.ts
Comment (ericrkuo) [reply src/services/PromotionService.ts:135]: Made a PR https://github.com/ubclaunchpad/PromoPal-backend/pull/137

Comment (ericrkuo) [reply src/services/PromotionService.ts:135]: You can use your admin privileges for the BE to merge the BE PR

Comment (andrewzulaybar) [reply src/services/PromotionService.ts:135]: Thank you!!

File: src/components/navigation/NavigationBar.tsx
Comment (andrewzulaybar) [reply src/components/navigation/NavigationBar.tsx:63]: Oh yes you're right, didn't notice that good catch! I'll move this

File: src/screens/MyPromotions.tsx
Comment (andrewzulaybar) [reply src/screens/MyPromotions.tsx:17]: Oh I caught this too no problem -- I asked Yu to take a look at that and he's working on it!

File: src/components/navigation/NavigationBar.tsx
Comment (andrewzulaybar) [reply src/components/navigation/NavigationBar.tsx:52]: Oh yeah you're right it happens for me too. Ahh I think that might be something to do with ant design... I'll have to figure this out another time I think :(

File: src/components/account/RegisterCard.tsx
Comment (andrewzulaybar) [reply src/components/account/RegisterCard.tsx]: Ah good catch thanks!

File: src/components/PromotionList.tsx
Comment (andrewzulaybar) [reply src/components/PromotionList.tsx:27]: Oh I don't remember seeing the discount for a promotion! Here's what the designs look like:

Screen Shot 2021-04-10 at 2 19 59 PM

File: src/screens/MyPromotions.tsx
Comment (ericrkuo) [reply src/screens/MyPromotions.tsx:17]: Wait actuallly I can fix this, making a PR for it rn

Comment (ericrkuo) [reply src/screens/MyPromotions.tsx:17]: Ohcrap

Comment (ericrkuo) [reply src/screens/MyPromotions.tsx:17]: I already started on it

File: src/components/PromotionList.tsx
Comment (ericrkuo) [reply src/components/PromotionList.tsx:27]: Hmm, it makes sense to show it right? How much effort would it take?

File: src/screens/MyPromotions.tsx
Comment (ericrkuo) [reply src/screens/MyPromotions.tsx:17]: BE PR https://github.com/ubclaunchpad/PromoPal-backend/pull/138

File: src/components/navigation/NavigationBar.tsx
Comment (ericrkuo) [reply src/components/navigation/NavigationBar.tsx:52]: Ah okay no worries

File: src/components/PromotionList.tsx
Comment (andrewzulaybar) [reply src/components/PromotionList.tsx:27]: @ericrkuo It shouldn't be too difficult, I think we might have to hold off on it for now because a concern from design was that the photo was too tall and that was because the length of the details in the promotion. I agree though it's a good idea to show so I've brought it up to them and they're still thinking about it!

Comment (ericrkuo) [reply src/components/PromotionList.tsx:27]: Ohh ok gotcah makes sense

**Review by ericrkuo — APPROVED** [review]

**Comment by ericrkuo** [comment]
Oops Im not sure where my comment went, I think github didnt add it.

AFter we merge, I would recommend doing full regression test of all our features just to be reaedy for demo. Test using our app when we are logged in vs llogged out and to try all other features.

The delete user thing is broken atm, Ive talked to Richard about it

**Comment by ericrkuo** [comment]
Also don'[t forget to change base branch 😄

**Pull request merged**
